.circleci/organise_doc.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const fs = require('fs-extra');
+const args = require('minimist')(process.argv.slice(2));
+
+// get SDK name and version from package files
+// they are used by JSDoc to generate the output directories
+const PACKAGE_JSON = require('../package.json');
+const PACKAGE_VERSION = PACKAGE_JSON.version;
+const PACKAGE_NAME = PACKAGE_JSON.name;
+
+// get the output directory
+const JSDOC_OUTPUT_DIR = require('../jsdoc.json').opts.destination;
+
+// root dir is the absolute path to use, provided by CI
+const ROOT_DIR = args.rootdir;
+// the tag being built and empty or null if none
+const TAG = args.tag;
+
+// if no tag, doc sill gon in latest directory
+const TARGET_DIR = (TAG == '' || TAG == null) ? 'latest' : TAG;
+
+// source directories where JSdoc put the doc
+const SOURCE_DIR_BASE = `${ROOT_DIR}/${JSDOC_OUTPUT_DIR}`;
+const SOURCE_DIR = `${SOURCE_DIR_BASE}/${PACKAGE_NAME}/${PACKAGE_VERSION}`;
+
+// check if source exists and move files to latest or tag matching directory
+fs.pathExists(SOURCE_DIR)
+.then(exists => {
```

Comment:
`.then` chaining makes your code harder to read... Also I don't think `pathExists` is a valid function in the `fs` module ([docs here](https://nodejs.org/api/fs.html))?

In this case I'd recommend just using `fs.existsSync` (returns bool) or `fs.statSync` (throws if file/dir doesn't exist). In the case of `fs.statSync` there's no need to wrap it in a try/catch block because you're just logging the exception to stdout, and letting it go uncaught will log it to stderr (which is where it should be written, anyway).

Otherwise, instead of promise chaining like this, it's way more readable to wrap the logic in an async function and use the `await` keyword.

So something like:

```javascript
someFuncThatReturnsPromise(...).then(retVal => { ... })
```

turns into
```javascript
const retVal = await someFuncThatReturnsPromise(...);
// use retVal here like you normally would in synchronous code
```

Reply:
oh, I missed that you were using `fs-extra` - in that case I'd recommend dropping that dependency and just use the basic `fs` module. It's easy enough to use in this case (you don't need async because you don't have any real concurrency), and it'll save installing an extra dependency, which will make the build faster.